p2p/conn_set.go

```diff
@@ -0,0 +1,86 @@
+package p2p
+
+import (
+	"net"
+	"sync"
+)
+
+// ConnSet is a lookup table for connections and all their ips.
+type ConnSet interface {
+	Get(string) net.Conn
```

Comment:
would be good to add argument name or godoc, so it's clear that one should pass `addr`

Reply:
also, Get and Set methods are asymmetric. Should probably be `Get(addr string)` and `Set(addr string, net.Conn, []net.IP)`. Alternatively, you can rename `Get` to `GetByAddr(string)`

Reply:
Removed the `Get` method completely as it is not used.